oprf/group/suite.go

```diff
@@ -0,0 +1,201 @@
+package group
+
+import (
+	"crypto/elliptic"
+	"crypto/rand"
+	"errors"
+	"io"
+	"math/big"
+
+	"github.com/armfazh/h2c-go-ref"
+)
+
+// Ciphersuite corresponds to the OPRF ciphersuite that is chosen. The
+// Ciphersuite object determines the prime-order group (pog) that is used for
+// performing the OPRF operations, along with the different hash function
+// definitions.
+// Should be created using NewSuite, using an string.
+type Ciphersuite struct {
+	// name of the ciphersuite
+	name string
+
+	// dst string used for hashing to curve
+	dst string
+
+	curve elliptic.Curve
+}
+
+// Suite is an interface that defines operations within additive
+// groups of prime order. This is the setting in which the OPRF operations
+// take place.
+type Suite interface {
+	// Returns the identifying name of the group
+	Name() string
+
+	// Returns the canonical (fixed) generator for defined group
+	Generator() *Point
+
+	// Returns the order of the canonical generator in the group.
+	Order() *Scalar
+
+	// Performs a transformation to encode bytes as a Element object in the
+	// group.
+	HashToGroup([]byte) (*Point, error)
+
+	// Performs a transformation to encode bytes as a Scalar object in the
+	// appropiate group.
+	// TODO: should we keep it? Seems only be used on the Verified case..
+	HashToScalar([]byte) (Scalar, error)
```

Comment:
Let's add it when the time comes. We can remove it for now. :)

Reply:
Ok ;)